pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -283,12 +283,22 @@ public PersistentTopic(String topic, ManagedLedger ledger, BrokerService brokerS
                 // ignore it for now and let the message dedup logic to take care of it
             } else {
                 final String subscriptionName = Codec.decode(cursor.getName());
-                subscriptions.put(subscriptionName, createPersistentSubscription(subscriptionName, cursor,
+                PersistentSubscription subscription = createPersistentSubscription(subscriptionName, cursor,
                         PersistentSubscription.isCursorFromReplicatedSubscription(cursor),
-                        cursor.getCursorProperties()));
-                // subscription-cursor gets activated by default: deactivate as there is no active subscription right
-                // now
-                subscriptions.get(subscriptionName).deactivateCursor();
+                        cursor.getCursorProperties());
+                subscriptions.put(subscriptionName, subscription);
+                subscription.getInitializeFuture()
+                        .exceptionally(t -> {
+                            log.warn("PersistentSubscription [{}] pendingAckHandleImpl relay failed "
+                                    + "when initialize topic [{}].", subscriptionName, topic, t);
+                            if (subscriptions.remove(subscriptionName, subscription)) {
+                                subscription.retryClose();
```

Comment:
why should add this method? persistentTopic init sub then remove will return false?

Reply:
I think it may happen. 
```
subscription.getInitializeFuture()
                        .exceptionally(t -> {
                              // ignore...
                        })
```
will be executed after the topic created. If a consumer connected the topic and unsubscribe the topic, `subscriptions.remove(...)` may return false.